.github/workflows/publish-docker-image.yml

```diff
@@ -2,57 +2,165 @@ name: Release multi-arch Docker Image
-      id: buildx
-      uses: docker/setup-buildx-action@v2
-
-    - name: Docker info
-      run: docker info
-    - name: Buildx inspect
-      run: docker buildx inspect
-
-    - name: Build and push image
-      uses: docker/build-push-action@v3
-      with:
-        context: .
-        file: ./Dockerfile
-        platforms: linux/amd64,linux/arm/v7,linux/arm64,linux/ppc64le,linux/s390x
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - run: go version
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: go-mod verify
+        run: go mod verify
+
+      - name: go-mod tidy
+        run: go mod tidy
+
+      - name: go-build
+        env:
+          GOOS: linux
+        run: go build
+
+      - name: go-test
+        run: go test -v ./...
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Docker info
+        run: docker info
+      - name: Buildx inspect
+        run: docker buildx inspect
+
+      - name: Build and push image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: linux/Dockerfile
+          platforms: linux/amd64,linux/arm/v7,linux/arm64,linux/ppc64le,linux/s390x,linux/386,linux/arm/v6
+          # Only push if we are publishing a release
+          push: true
+          tags: ${{ env.IMAGE_REPOSITORY }}:${{ github.event.release.tag_name }}-linux
+
+  release-windows:
+    strategy:
+      matrix:
+        os-version:
+          - ltsc2019
+          - ltsc2022
+    runs-on: windows-2022
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+      - run: go version
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: go-mod verify
+        run: go mod verify
+
+      - name: go-mod tidy
+        run: go mod tidy
+
+      - name: go-build
+        env:
+          GOOS: windows
+        run: go build
+
+      #- name: go-test
+      #  run: go test -v ./...
+
+      - name: Docker info
+        run: docker info
+
+      - name: Build image
+        run: |
+          docker build -f windows/Dockerfile --build-arg BASE_IMAGE=mcr.microsoft.com/windows/nanoserver:${{ matrix.os-version }} -t ${{ env.IMAGE_REPOSITORY }}:${{ inputs.tag_name }}-windows.amd64.${{ matrix.os-version }} .
+
```

Comment:
One thing I would like to propose is to stick to the same naming convention that MS uses, e.g.: `8.0.201-nanoserver-ltsc2022`.

Reply:
The 2nd one `testcontainers/ryuk:${{tag}}-nanoserver-${{ matrix.os-version }}`.

Reply:
I expected that the `windows` infix would be more expressive from the moby-ryuk perspective to distinct between linux and windows. Whether it's a nanoserver base image or any other one is more of an implementation detail. From Microsoft's perspective it's clear, that a "windows" would be to generic. I'm fine with both variants.
That said: I'll leave such a decision to the maintainers and yes, that's something we can easily fix afterwards, because it won't affect any users of the multi-arch image.

Reply:
> I'll leave such a decision to the maintainers and yes, that's something we can easily fix afterwards, because it won't affect any users of the multi-arch image.

Let's leave it as it is and not postpone the PR any longer. I do not want to end up in a situation similar to the previous PR too 😄.